examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -0,0 +1,125 @@
+"""
+=======================================================================
+Balance model complexity and cross-validated score using refit=callable
+=======================================================================
+
+A simple example demonstrates the usage of `refit=callable` interface in 
+`sklearn.model_selection.GridSearchCV`. It shows this interface adds certain
+amount of flexibility in identifying the "best" estimator. The function 
+passed to paramter `refit` incorporate of which metric(s) to optimise. This
+interface can also be used in multiple metrics evaluation.
+
+This example balances model complexity and cross-validated score by 
+finding a decent accuracy within 1 standard deviation of the best accuracy
```

Comment:
You might want to say that this is a rule of thumb for insignificant difference.

We could determine insignificant difference in a more proper way, such as with a wilcoxon rank-sum test